languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml

```diff
@@ -2909,8 +2909,28 @@ USA
                 <token spacebefore="no">i</token>
             </antipattern>
             <rule>
+                <!-- antipatterns intended for mathematical or scientific usages of the symbol "i" -->
                 <antipattern>
-                    <token regexp="yes">row|index</token>
+                    <token regexp="yes">(row|index|variable|quantit(y|ie)|number|term|symbol|constant|point|element|unit|value|size|equal)s?|sub|negative|minus|lowercase</token>
+                    <token>i</token>
```

Comment:
To make this AP a bit less zealous, maybe you could add

`<exception scope="next" postag="VB[PD]" postag_regexp="yes"/>`

That way, we'll still find mistakes like

_This is not the quantity i ordered._
_He didn't even try to understand the point i wanted to make._
_I'm going to show you how negative i can be._